### PR TITLE
AmbigNQ for ReAct

### PR DESCRIPTION
**Pull request by tedasdf**
The AmbigQA benchmark is a dataset and evaluation setup designed to test a machine learning model's ability to comprehend and generate answers to ambiguous questions. It contains questions that have multiple valid interpretations or answers, challenging models to understand the context and provide nuanced responses.

### 🤔 Reasoning

_Explain the purpose of this PR..._

### 🚧 Changes

- Add prompt 
- Demo

### ✅ PR Checklist

- [x] Using this PR template?
- [x] Linked issue?
- [x] Added feature?
    - [x] Added/updated docs?
    - [x] Added/updated tests?

### CODE REVIEW

File: agential/cog/prompts/react.py [hunk 1]
```diff
@@ -118,3 +118,61 @@
...
+You have a maximum of {max_steps} steps.
+
+Here are some examples:
+{examples}
+(END OF EXAMPLES)
+
+Question: {question}{scratchpad}"""
+
+REACT_AMBIGQA_EXAMPLES = """
+Question: When did men's figure skating become a summer Olympic sport?
```
Comment (alckasoc) [inline agential/cog/prompts/react.py:138]:
```suggestion
REACT_AMBIGQA_EXAMPLES = """Question: When did men's figure skating become a summer Olympic sport? 
```

File: agential/cog/prompts/react.py [hunk 2]
```diff
@@ -118,3 +118,61 @@
...
+Question: Who plays wilmas mom in the flintstones movie? 
+Thought 1: Let's search the question in google 
+Action 1: Search[Who played Wilma's mom in the flintstones movie? site: wikipedia.org] 
+Observation 1: [Elizabeth Perkins - Wikipedia] Elizabeth Ann Perkins (born November 18, 1960 ) is an American actress. 
+Thought 2: The evidence does not mention Wilma's mom in the flintstones movie. 
+Action 2: Search[Who played Wilma's mom in the flintstones movie?] 
+Observation 2: [] The Flintstones / Wilma Flintstone / Mother / Played by Elizabeth Taylor 
+Thought 3: The evidence shows that Elizabeth Taylor played the role of Wilma's mother. 
+Action 3: Finish[Elizabeth Taylor]
+"""
```
Comment (alckasoc) [inline agential/cog/prompts/react.py:178]:
```suggestion
Action 3: Finish[Elizabeth Taylor]"""
```

**Review by alckasoc — CHANGES_REQUESTED** [review]
could you add print statements within the `_prompt_*` functions to print out the `prompt` and the `out`?

**Comment by jsgage7** [comment]
@tedasdf Prompts and example LGTM. Please update the PR description to be more descriptive and check the boxes, including linking the issue. I see you already completed the to-do boxes on the issue itself, thanks for doing that! Also looks like there are some merge conflicts that need to be resolved.

@alckasoc Just wanted to clarify on the process for the tmp.ipynb. In my previous PR, I created a separate branch so that I could link to the committed notebook without affecting the PR. Not sure if there's a better way to do this, maybe testing it locally and posting screenshots without committing the notebook? But seems like this will be a common thing so it would be good to have best practices on it.

**Comment by alckasoc** [comment]
hmmmmm

**Comment by alckasoc** [comment]
yea i guess we could test it locally without committing the nb + maybe upload the nb to slack?

**Comment by jsgage7** [comment]
I notice that all messages being sent in this chat are appearing as comments on the PR, I'm going to attach a notebook to this message in slack and see what happens on GH

[Slack File: test.ipynb](https://files.slack.com/files-pri/T06G9G0703E-F071HFXMBT6/test.ipynb)

**Comment by alckasoc** [comment]
it works! :sunglasses:

**Comment by jsgage7** [comment]
Okay so @tedasdf if you could make the changes I mentioned and uncommit the notebook, you should be good to go

**Comment by tedasdf** [comment]
I uncommit the notebook and resovle the conflict . should i commit merge?

**Review by alckasoc — APPROVED** [review]

**Pull request merged**